webrtc/README.md

```diff
@@ -0,0 +1,412 @@
+  attacks?
+
+    - Can an attacker launch an **amplification attack** with the STUN endpoint
+      of the server?
+
+      The QUIC protocol defends against amplification attacks by requiring:
+
+      > an endpoint MUST limit the amount of data it sends to the unvalidated
+      > address to three times the amount of data received from that address.
+
+      https://datatracker.ietf.org/doc/html/rfc9000#section-8
+
+      For WebRTC in libp2p one could require the client (_A_) to add additional
+      bytes to its STUN message, e.g. in the STUN username and password, thus
+      making an amplification attack less attractive.
+
+    - Can a client run a **DOS attack** by sending many STUN messages with
+      different ufrags using different UDP source ports, forcing the server to
+      allocate a new peer connection for each? Would rate limiting suffice to
+      defend against this attack?
+
+  See also:
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.2.1
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.1.2
+
+- Do the major WebRTC server implementations support using the same UDP port for
+  multiple WebRTC connections, thus requiring multiplexing multiple WebRTC
+  connections on the same UDP port?
+
+  This is related to [ICE Lite](https://www.rfc-editor.org/rfc/rfc5245), having
+  a host only advertise a single address, namely the host address, which is
+  assumed to be public.
+
+  The [Go WebRTC](https://github.com/pion/webrtc/) implementation and the [Rust
+  WebRTC](https://github.com/webrtc-rs/webrtc) implementation support this. It
+  is unclear whether this is supported in any of the NodeJS implementations.
+
+- Do the major (Go / Rust / ...) WebRTC implementations allow us to accept a
+  WebRTC connection from a remote node without previously receiving an SDP
+  packet from such host?
+
+- Can _Browser_ generate a _valid_ SDP packet for the remote node based on the
+  remote's Multiaddr, where that Multiaddr contains the IP, UDP port and TLS
+  certificate fingerprint (e.g.
+  `/ip6/2001:db8::/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`)? _Valid_ in
+  the sense that this generated SDP packet can then be used to establish a
+  WebRTC connection to the remote.
+
+  Yes.
+
+### Browser to Browser
+
+Scenario: Browser _A_ wants to connect to Browser node _B_ with the help of
+server node _R_.
+
+- Replace STUN with libp2p's identify and AutoNAT
+  - https://github.com/libp2p/specs/tree/master/identify
+  - https://github.com/libp2p/specs/blob/master/autonat/README.md
+- Replace TURN with libp2p's Circuit Relay v2
+  - https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md
+- Use DCUtR over Circuit Relay v2 to transmit SDP information
+  1. Transform ICE candidates in SDP to multiaddresses.
+  2. Transmit the set of multiaddresses to the remote via DCUtR.
+  3. Transform the set of multiaddresses back to the remotes SDP.
```

Comment:
Can we get rid of the two transformation steps by transmitting the SDP directly.